internal/pkg/agent/application/info/svc_windows.go

```diff
@@ -7,44 +7,65 @@
 package info
 
 import (
-	"golang.org/x/sys/windows"
-)
+	"fmt"
 
-const (
-	ML_SYSTEM_RID = 0x4000
+	"golang.org/x/sys/windows"
 )
 
 // RunningUnderSupervisor returns true when executing Agent is running under
 // the supervisor processes of the OS.
+//
+// Checks in the following order:
+//  1. Has SECURITY_LOCAL_SYSTEM_RID (aka. running as LOCAL SYSTEM)
+//  2. Has SECURITY_SERVICE_RID (aka. running as service as non LOCAL SYSTEM user)
 func RunningUnderSupervisor() bool {
-	serviceSid, err := allocSid(ML_SYSTEM_RID)
+	localSystem, _ := hasLocalSystemSID()
+	if localSystem {
+		return true
+	}
+	isService, _ := hasServiceSID()
+	return isService
+}
+
+func hasLocalSystemSID() (bool, error) {
+	sid, err := allocSid(windows.SECURITY_LOCAL_SYSTEM_RID)
 	if err != nil {
-		return false
+		return false, fmt.Errorf("allocate sid error: %w", err)
 	}
-	defer windows.FreeSid(serviceSid)
+	defer func() {
+		_ = windows.FreeSid(sid)
+	}()
 
-	t, err := windows.OpenCurrentProcessToken()
+	token := windows.Token(0)
```

Comment:
`windows.Token(0)` is used because its passed into `checkTokenMembership` that when the token is zero that check uses the current process token. It is a simpler way of saying this processes current token, without the need to grab a handle that needs to be closed for the current process.

What do you want documented here? That `windows.Token(0)` means the current process token?

Reply:
> What do you want documented here? That `windows.Token(0)` means the current process token?

Probably a few links to MSDN documentation, and just a comment that `windows.Token(0)` is current process.  I'm just thinking in 3 years if we have a bug around this code in some new Azure environment it is nice to know what we were thinking and hopefully pointers to full documentation which might give clues if it has been deprecated etc.

Reply:
Added some more comments and a link for the `windows.Token(0)`.